subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5663,7 +5663,7 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 #endif /* CONFIG_LWM2M_DTLS_SUPPORT */
 
 	if (connect(client_ctx->sock_fd, &client_ctx->remote_addr,
-		    NET_SOCKADDR_MAX_SIZE) < 0) {
+	        client_ctx->addr_len) < 0) {
```

Comment:
Perhaps, instead of storing the address size in the context, just resolve the size here based on `client_ctx->remote_addr.sa_family`?